code/modules/projectiles/updated_projectiles/guns/rifles.dm

```diff
@@ -421,3 +421,45 @@
 	recoil_unwielded = config.high_recoil_value
 
 //-------------------------------------------------------
+//MK224
+
+/obj/item/weapon/gun/rifle/mk224
+	name = "\improper MK224 Tactical Shotgun"
+	desc = "A modified version of the MK221 Tactical Shotgun, now including a magazine."
+	icon_state = "mk224"
+	item_state = "mk221"
+	fire_sound = "sound/weapons/gun_shotgun_automatic.ogg"
+	current_mag = /obj/item/ammo_magazine/rifle/mk224/slug
+	attachable_allowed = list(
+						/obj/item/attachable/quickfire,
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/stock/tactical,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/scope/mini)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER
+	starting_attachment_types = list(/obj/item/attachable/attached_gun/grenade)
+
+/obj/item/weapon/gun/rifle/mk224/New()
+	. = ..()
+	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 19,"rail_x" = 10, "rail_y" = 21, "under_x" = 24, "under_y" = 16, "stock_x" = 14, "stock_y" = 16)
```

Comment:
Move this var to the path instead of New(). This was never needed and has been fixed on other guns too.